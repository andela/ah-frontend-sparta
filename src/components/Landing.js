import React from 'react';
import { Button } from 'react-toolbox/lib/button';

const Landing = () => (
  <>
    <h1>Landing Page</h1>
    <Button label="Bookmark" raised primary />
  </>
);

export default Landing;
