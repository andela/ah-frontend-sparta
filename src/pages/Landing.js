import React from 'react';
import Navbar from '../components/Navbar';
import LandingPageWrapper from '../components/LandingPageWrapper';
import LandingImage from '../../images/landing_image.jpg';

import '../assets/scss/style.scss';


const Landing = () => (
    <div>
      <Navbar/>
      
      <div className="page-banner-image">
         <img className='LandingImage' src={LandingImage} alt="Landing image" />
      </div>
      <div className="small-screen">
        <div>
            <form>
              <input type="text" placeholder="SEARCH" />
            </form>
        </div>
        <div><a href="/login">Account</a></div>
      </div>
      <LandingPageWrapper  />
      <br></br>
      <br></br>
      <br></br>
    </div>
    
);

export default Landing;
