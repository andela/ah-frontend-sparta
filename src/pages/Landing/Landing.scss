@import '../../assets/scss/shared.scss';
$heading: green;

//h1 heading color
h1 {
  color: $heading;
}

.flex-container {
  display: flex;
  background-color: DodgerBlue;

  justify-content: center;
  height: 100%;
  align-items: stretch;


  flex-direction: row;
  div {
    width: rem(720);

  }

  .login-div {
    background-color:#172439;
    height: 52em;
  }

  .section_one{
    height: 52em;
    background-size: cover;
  }
}

.content {
  align-items: center;
  background: rgb(0, 0, 0); /* Fallback color */
  background: rgba(0, 0, 0, 0.5); /* Black background with 0.5 opacity */
  color: #f1f1f1;
  width: rem(340);
  height: rem(790);
  padding: rem(20);
  font-size: rem(25);

  div {
    margin-top: 3em;
    width: rem(670);
    text-align: center;
  }
}

.social_icons {
   display: flex;
   flex-direction: row;
   justify-content: flex-end;

   div {
    background-color: #f1f1f1;
    margin: rem(10);
    padding: rem(20);
    width: rem(10);
    height: rem(10);
    font-size: rem(30);
   }
}

.login_signup_button {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;

}

@media (max-width: 576px) {
   .flex-container {
     display: flex;
     flex-direction: column;
     div {
       background-color: #f1f1f1;
       width: rem(340);
       height: rem(295);
       border: rem(2) solid red; }
   }

}

.navbar_container {
    display: flex;
    background-color: #172439;

    div  {
      background-color: red;
      padding: rem(10);
      margin: rem(10);
    }
}

.menu-container {
  background-color: #172439;
  padding: rem(20) 0;
  display: flex;

  .menu {
    border: rem(1) solid #fff;  /* For debugging */
    width: rem(900);
  }
}

// navbar css

*{
  padding: 0px;
  margin: 0px;
  box-sizing: border-box;
}

*{
  padding: 0px;
  margin: 0px;
  box-sizing: border-box;
}

.nav{
  background: #172439;
  color: white;
  height: rem(106);
  font-family: sans-serif;
  display: flex;
  flex-wrap:wrap;
  align-items: center;
  border-bottom: rem(1) solid gray;

  .nav-logo {
    width: 20%;
    // background-color: red;
    padding: 16px 40px;
    text-align: center;

    img {
      height: rem(70);
      width: rem(150);
    }
  }
  .nav-menu {
    width: 80%;
  }
  .nav-menu {
    ul {
      display: flex;
      justify-content: flex-end;
    }
  }
  .nav-menu {
    ul {
      li {
        padding: rem(10) rem(20);
        color:#ccc;
        font-size: rem(20);
        list-style-type: none;
        cursor: pointer; 
      }
    }
  }
  .nav-menu {
    ul {
      li:hover {
        color:#fff;
        font-weight: bold;
      }
    }
  }
  .nav-menu {
    ul {
      li{
        a {
          text-decoration: none;
          color: #fff;
        }
      }
    }
  }

  .nav-menu {
    ul {
      li {
        form {
          input {
            outline: 0;
            border-width: 0 0 2px;
            border-color: #fff;
            width: rem(200);
            height: rem(30);
            background-color: #172439;
            color:#fff;
            font-size: rem(16);
          }
        }
      }
    }
  }
}
// change searchbox color to white
::placeholder { 
  color:#fff;
  opacity: 1; 
}

// Changing the width and height of an image in the landing page 
.LandingImage {
  height: rem(400);
  width: 100%;
}

.wrapper {
  display: grid;
  min-height: rem(100);
  grid-template-columns: 2fr 1fr;
  padding-top: rem(30);
  margin-left: rem(120);
  margin-right: rem(120);
  grid-column-gap: 3em;
  grid-auto-rows:minmax(rem(100), auto);

  div {
      .sample-articles {
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-auto-rows: 9em;
    grid-gap: 1em;
  
    .article-image {
      img {
        height: rem(140);
        width: rem(214);
      }
   }
    .sample_article_start {
      color:  #6A6A6A;
      text-align: justify;
    }
    
  }

  }
  div {
    // border: 1px solid red;

    h1 {
      font-family: Roboto;
      font-style: normal;
      font-weight: bold;
      font-size: rem(25);
      line-height: 117%;
      text-align: center;
      letter-spacing: 0.03em;
      color: #7C7B7B;
    }

    h4 {
        font-family: Roboto;
        font-style: normal;
        font-weight: bold;
        font-size: rem(20);
        letter-spacing: 0.03em;
        color: #000000;
        text-align: justify;
    }
    .author_name {
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: rem(18);
      letter-spacing: 0.03em;
      text-align: justify;
      color: #000000;
    }
    h6 {
      font-family: Roboto;
      font-style: normal;
      font-weight: normal;
      font-size: rem(16);
      line-height: 117%;
      letter-spacing: 0.03em;
      text-align: justify;
      color: #6A6A6A;
    }
  }
}

.trending {
  display: grid;
  grid-template-columns:80% 10%;
  grid-auto-rows: 4em;
  padding-left: rem(10);

  div {

    img{
      height: rem(20);
      width: rem(20);
    }
  }
}

.book-mark-holder {
   text-align: center;
   vertical-align: middle;
   line-height: rem(100);
}



.small-screen {
  display: none;
}


@media screen and (max-width:576px) {

   .wrapper {
        display: grid;
        grid-template-rows: 60% 50%;
        padding-top: rem(30);
        margin-left: rem(40);
        margin-right: rem(40);
        grid-column-gap: 3em;
        grid-auto-rows: minmax(rem(100), auto);
        grid-template-columns: none !important;

      div {
        .sample-articles {
          padding-top:rem(10);
          display: grid !important;
          grid-template-rows: 50% !important;
          grid-auto-columns: minmax(rem(300px), auto);
          grid-template-columns: none !important;

          .article-image img {
            height: rem(180);
            width: rem(285);
           }
           .trending-articles {
             h4 {
               text-align: none !important;
             }
             h6 {
               line-height: 2em;
             }
           }
      }

      .trending {
        display: grid;
        grid-template-columns: 80% 10%;
        grid-auto-rows: 6em;
        padding-left: 0px !important;
        justify-content: center;
      }

     }
      
   }
   .small-screen {
     display: grid;
     grid-template-columns: 2fr 1fr;
     grid-auto-rows: 4em;

     div {
       text-align: center;
       padding-top: rem(20);
     
       a {
         text-decoration: none;
         font-style: bold;
         color: #000;
       }

       form {
        input {
          outline: 0;
          border-width: 0 0 rem(2);
          border-color: #fff;
          width: rem(200);
          height: rem(30);
          background-color: #172439;
          color:#fff;
          font-size: rem(16);
        }
      }
     
     }
    
   }

   .nav {
     height: auto;

     .nav-logo {
       padding-top: rem(70);
       height:rem(200);
       width:100%;
    
     }
     .nav-menu {
       ul {
        width: 100%;
        justify-content: space-around;
       }
     }
   }
   .nav {
     .nav-menu {
       ul{
         li {
           display: none;
           width: 20%;
           padding: rem(10) rem(5);
           text-align: center;
           font-size: rem(16);
         }
       }
     }
   }
   .nav {
    .nav-menu {
      width: 100%;
    }
  }

  .page-banner-image {
    display: none;
  }
  
 
}

