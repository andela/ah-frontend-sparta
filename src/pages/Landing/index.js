import React from 'react';
import LandingPageWrapper from '../../components/LandingPageWrapper';
import LandingImage from '../../assets/images/landing_image.jpg';

import './Landing.scss';


const Landing = () => (
    <div>
      <div className="page-banner-image">
         <img className='LandingImage' src={LandingImage} alt="Landing image" />
      </div>
      <div className="small-screen">
        <div>
            <form>
              <input type="text" placeholder="SEARCH" />
            </form>
        </div>
        <div><a href="/login">Account</a></div>
      </div>
      <LandingPageWrapper  />
      <br></br>
      <br></br>
      <br></br>
    </div>
    
);

export default Landing;
