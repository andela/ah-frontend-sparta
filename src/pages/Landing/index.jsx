import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ArticleContainer from '../../components/Articles/ArticleContainer';
import LandingImage from '../../assets/images/landing_image.jpg';
import { fetchArticles } from '../../actions/articlesActions';
import { getOriginal, getNext } from '../../actions/paginationActions';
import './Landing.scss';

export class Landing extends Component {
  componentDidMount() {
    const { getArticles, fetchOriginal } = this.props;
    getArticles();
    fetchOriginal();
  }

  fetchData = (event) => {
    event.preventDefault();
    const { fetchNext } = this.props;
    fetchNext(localStorage.next);
  };

  fetchPrevious = (event) => {
    event.preventDefault();
    const { fetchNext } = this.props;
    fetchNext(localStorage.previous);
  };

  render() {
    const { firstarticles, paginatearticles, pageState: { next, prev } } = this.props;
    return (
      <div>
        <div className="page-banner-image">
          <img className="LandingImage" src={LandingImage} alt="Landing" />
        </div>
        <div className="small-screen">
          <div>
            <form>
              <input type="text" placeholder="SEARCH" />
            </form>
          </div>
          <div>
            <a href="/login">Account</a>
          </div>
        </div>
        <div>
          <div>
            {paginatearticles.length === 0 ? <ArticleContainer articles={firstarticles} /> : <ArticleContainer articles={paginatearticles} />}
            <br />
            <br />
          </div>
        </div>
        <div className="container pagination-buttons">
          {prev ? (
            <button type="button" className="btn left pagButton" onClick={this.fetchPrevious}>
              Previous
            </button>
          ) : (
            <button type="button" className="btn left pagButton" disabled>
                Previous
            </button>
          )}
          {next ? (
            <button type="button" className="btn right pagButton" onClick={this.fetchData}>
              Next
            </button>
          ) : (
            <button type="button" className="btn right pagButton" disabled>
                Next
            </button>
          )}
        </div>
      </div>
    );
  }
}

Landing.defaultProps = {
  firstarticles: [],
  paginatearticles: [],
};
Landing.propTypes = {
  firstarticles: PropTypes.arrayOf(PropTypes.object),
  paginatearticles: PropTypes.arrayOf(PropTypes.object),
  getArticles: PropTypes.func.isRequired,
  fetchOriginal: PropTypes.func.isRequired,
  fetchNext: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  paginatearticles: state.paginateArticles,
  firstarticles: state.firstArticles,
  pageState: state.pageState,
});
const mapDispatchToProps = dispatch => ({
  fetchOriginal: () => {
    dispatch(getOriginal());
  },
  fetchNext: (url) => {
    dispatch(getNext(url));
  },
  getArticles: () => {
    dispatch(fetchArticles());
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(Landing);
