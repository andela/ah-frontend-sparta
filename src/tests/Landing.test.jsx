import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Landing from '../pages/Landing';

describe('Landing page', () => {
  it('should render without crashing', () => {
    const wrapper = renderer.create(<Router><Landing /></Router>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
