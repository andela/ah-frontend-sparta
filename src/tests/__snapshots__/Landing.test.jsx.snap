// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page should render without crashing 1`] = `
<div>
  <div
    className="page-banner-image"
  >
    <img
      alt="Landing page."
      className="LandingImage"
      src={Object {}}
    />
  </div>
  <div
    className="small-screen"
  >
    <div>
      <form>
        <input
          placeholder="SEARCH"
          type="text"
        />
      </form>
    </div>
    <div>
      <a
        href="/login"
      >
        Account
      </a>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post one
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              Roghashin Timbiti
            </b>
          </p>
          <h6>
            May 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post two
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              null null
            </b>
          </p>
          <h6>
            April 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post three
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              null null
            </b>
          </p>
          <h6>
            April 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post four
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              null null
            </b>
          </p>
          <h6>
            April 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post five
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              null null
            </b>
          </p>
          <h6>
            April 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post six
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              null null
            </b>
          </p>
          <h6>
            April 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
      <div
        className="sample-articles"
      >
        <div
          className="article-image"
        >
          <img
            alt="Article"
            src={Object {}}
          />
        </div>
        <div
          className="trending-articles"
        >
          <h4>
            Post seven
          </h4>
          <h6
            className="sample_article_start"
          >
            too long a story
          </h6>
          <p
            className="author_name"
          >
            <b>
              null null
            </b>
          </p>
          <h6>
            April 18, 2019
             
            - 3 min read
          </h6>
        </div>
      </div>
    </div>
    <div>
      <h1>
        Trending
      </h1>
      <br />
      <hr />
      <br />
      <div
        className="trending"
      >
        <div
          className="trending-articles"
        >
          <h4>
            Why do we sleep
          </h4>
          <p
            className="author_name"
          >
            By Francis Kiryowa
          </p>
          <h6>
            APR 27, 2019 - 3 min read
          </h6>
        </div>
        <div
          className="book-mark-holder"
        >
          <img
            alt="bookmark"
            className="bookmark-image"
            src={Object {}}
          />
        </div>
      </div>
      <br />
      <div
        className="trending"
      >
        <div
          className="trending-articles"
        >
          <h4>
            Why do we sleep
          </h4>
          <p
            className="author_name"
          >
            By Francis Kiryowa
          </p>
          <h6>
            APR 27, 2019 - 3 min read
          </h6>
        </div>
        <div
          className="book-mark-holder"
        >
          <img
            alt="bookmark"
            className="bookmark-image"
            src={Object {}}
          />
        </div>
      </div>
      <br />
      <div
        className="trending"
      >
        <div
          className="trending-articles"
        >
          <h4>
            Why do we sleep
          </h4>
          <p
            className="author_name"
          >
            By Francis Kiryowa
          </p>
          <h6>
            APR 27, 2019 - 3 min read
          </h6>
        </div>
        <div
          className="book-mark-holder"
        >
          <img
            alt="bookmark"
            className="bookmark-image"
            src={Object {}}
          />
        </div>
      </div>
      <br />
      <div
        className="trending"
      >
        <div
          className="trending-articles"
        >
          <h4>
            Why do we sleep
          </h4>
          <p
            className="author_name"
          >
            By Francis Kiryowa
          </p>
          <h6>
            APR 27, 2019 - 3 min read
          </h6>
        </div>
        <div
          className="book-mark-holder"
        >
          <img
            alt="bookmark"
            className="bookmark-image"
            src={Object {}}
          />
        </div>
      </div>
    </div>
  </div>
  <br />
  <br />
  <br />
</div>
`;
